### PR TITLE
flattened out that do/while loop that only ran once in DataChannelRer…

### PR DESCRIPTION
**Pull request by dwalend**

### CODE REVIEW

File: src/main/java/ev3dev/utils/DataChannelRereader.java [hunk 1]
```diff
@@ -50,15 +50,14 @@ public DataChannelRereader(String pathString) {
...
-                channel.position(0);
-                n = channel.read(byteBuffer);
-                if (n == -1) {
-                    throw new IOException("Premature end of file ");
-                }
-            } while (n <= 0);
+            byteBuffer.clear();
+            channel.position(0);
+            int n = channel.read(byteBuffer);
+            if (n == -1) {
```
Comment (JakubVanek) [inline src/main/java/ev3dev/utils/DataChannelRereader.java:56]: I think this branch should handle the `n == 0` case as well. I'm not sure when would this happen, but if it happens then the code below will break (`bytes[-1]` is invalid).

File: src/main/java/ev3dev/utils/DataChannelRereader.java [hunk 2]
```diff
@@ -50,15 +50,14 @@ public DataChannelRereader(String pathString) {
...
-                byteBuffer.clear();
-                channel.position(0);
-                n = channel.read(byteBuffer);
-                if (n == -1) {
-                    throw new IOException("Premature end of file ");
-                }
-            } while (n <= 0);
+            byteBuffer.clear();
+            channel.position(0);
+            int n = channel.read(byteBuffer);
```
Comment (JakubVanek) [inline src/main/java/ev3dev/utils/DataChannelRereader.java:55]: The `position()` + `read()` calls can be merged together: https://github.com/ev3dev-lang-java/ev3dev-lang-java/blob/1a18470748c1fca44afd342acb37c4335e654774/src/main/java/ev3dev/utils/DataChannelRereader.java#L54 With that change, all FileChannel accesses should be thread-safe (I think including close), only the buffer would be unsafe.

File: src/main/java/ev3dev/utils/DataChannelRereader.java (hunk 1 above)
Comment (jabrena) [reply src/main/java/ev3dev/utils/DataChannelRereader.java]: It is not a bad idea.
I will merge this version and later we will refine with that idea

Comment (JakubVanek) [reply src/main/java/ev3dev/utils/DataChannelRereader.java]: This was already added: https://github.com/ev3dev-lang-java/ev3dev-lang-java/pull/778/commits/c1b78979bd5d9dd4eb638b855153f2ec42fcb868 (later commit in this PR)

**Pull request merged**